hydra-node/src/Hydra/Chain/Direct.hs

```diff
@@ -253,6 +253,14 @@ newtype IntersectionNotFoundException = IntersectionNotFound
 
 instance Exception IntersectionNotFoundException
 
+data ChainClientException
+  = EraNotSupportedException
+  deriving stock (Show)
+
+instance Exception ChainClientException where
+  displayException = \case
+    EraNotSupportedException -> "Received blocks in unsupported era. Please upgrade your hydra-node."
```

Comment:
Should: mention which era we saw in the message

Reply:
Unfortunately for this case we cannot get this information easily from the context, thus I had to hardcode the era missmatch.

wdyt?